### ISSUE TITLE
How to ag-grid cellrender boolean checkbox?

### ISSUE THREAD

**Issue by knoxvilledatabase** [elimintz/justpy#114]
How would I render a boolean field to checkboxes? Not for selecting the row, only for managing the value of that cell?

![Screen Shot 2020-07-29 at 4 51 33 AM](https://user-images.githubusercontent.com/2231079/88778993-3765cd80-d157-11ea-8633-90f915e19431.png)

This is what I tried: 

```python
import justpy as jp

grid_options = """
{
    defaultColDef: {
        filter: true,
        sortable: true,
        resizable: true,
        cellStyle: {textAlign: 'center'},
        headerClass: 'font-bold'
    }, 
      columnDefs: [
      {headerName: "Make", field: "make"},
      {headerName: "Model", field: "model"},
      {headerName: "Price", field: "price"},
      {headerName: "Enabled", field: "enabled"}
    ],
      rowData: [
      {make: "Toyota", model: "Celica", price: 35000, enabled: true},
      {make: "Ford", model: "Mondeo", price: 32000, enabled: false},
      {make: "Porsche", model: "Boxter", price: 72000, enabled: false}
    ]
}
"""

def grid_test():
    wp = jp.WebPage()
    grid = jp.AgGrid(a=wp, options=grid_options)
    grid.options.columnDefs[3] = {
        'cellRenderer': 'checkboxRenderer',
        'headerName': "Enabled",
        'field': "enabled"
    }
    return wp

jp.justpy(grid_test)
```

**Comment by snoodlenay** [comment]
There's a way to render html in a cell. Is this what you're looking for?

```
grid_options = """
{
    defaultColDef: {
        filter: true,
        sortable: true,
        resizable: true,
        cellStyle: {textAlign: 'center'},
        headerClass: 'font-bold'
    }, 
      columnDefs: [
      {headerName: "Make", field: "make"},
      {headerName: "Model", field: "model"},
      {headerName: "Price", field: "price"},
      {headerName: "Enabled", field: "enabled"}
    ],
      rowData: [
      {make: "Toyota", model: "Celica", price: 35000, enabled: '<input type="checkbox" checked="checked">'},
      {make: "Ford", model: "Mondeo", price: 32000, enabled: '<input type="checkbox">'},
      {make: "Porsche", model: "Boxter", price: 72000, enabled: '<input type="checkbox">'}
    ]
}
"""

def grid_test():
    wp = jp.WebPage()
    grid = jp.AgGrid(a=wp, options=grid_options)
    grid.options.columnDefs[3] = {
        'cellRenderer': 'checkboxRenderer',
        'headerName': "Enabled",
        'field': "enabled"
    }
    grid.html_columns = [3]
    return wp
```

**Comment by elimintz** [comment]
I am out of the office, so it will take me a few days to find a good solution for this.

Please take a look at https://github.com/elimintz/justpy/issues/107

It is easy enough to render the check boxes:
```python

import justpy as jp

grid_options = """
{
    defaultColDef: {
        filter: true,
        sortable: true,
        resizable: true,
        cellStyle: {textAlign: 'center'},
        headerClass: 'font-bold'
    }, 
      columnDefs: [
      {headerName: "Make", field: "make"},
      {headerName: "Model", field: "model"},
      {headerName: "Price", field: "price"},
      {headerName: "Enabled", field: "enabled"}
    ],
      rowData: [
      {make: "Toyota", model: "Celica", price: 35000, enabled: '<input id="check1" type="checkbox" onclick="{console.log(this); console.log(this.checked)}" />'},
      {make: "Ford", model: "Mondeo", price: 32000, enabled: '<input id="check2" type="checkbox" />'},
      {make: "Porsche", model: "Boxter", price: 72000, enabled: '<input id="check3" type="checkbox" />'}
    ]
}
"""


def grid_test():
    wp = jp.WebPage()
    grid = jp.AgGrid(a=wp, options=grid_options)
    grid.html_columns = [3]
    return wp

jp.justpy(grid_test)

```

The question remains how to get the value of the click back to the server. Probably will have to add to  the component JavaScript code.

**Comment by knoxvilledatabase** [comment]
Thank you! Exactly, I'm trying to get the value back to the server so I print it in python.

**Comment by knoxvilledatabase** [comment]
@elimintz, ag-grid, Vue checkbox. Does this help? 

https://plnkr.co/edit/KIlOopnaJUrg0c7A?preview

Link to ag-grid blog: https://blog.ag-grid.com/binding-boolean-values-to-checkboxes-in-ag-grid/

**Comment by elimintz** [comment]
Yes, this is helpful. It currently looks like the best option. In any case this means making changes also in the JavaScript so will require a new version.

**Comment by elimintz** [comment]
This feature has been added in the just released version 0.1.4:
https://justpy.io/grids_tutorial/checkbox/
